In [12]:
import pandas as pd
import matplotlib.pyplot as plt
import datetime
df=pd.read_csv('US CPI.csv')
df


Date      CPI
0     1/1/1913    9.800
1     1/2/1913    9.800
2     1/3/1913    9.800
3     1/4/1913    9.800
4     1/5/1913    9.700
...        ...      ...
1298  1/3/2021  264.877
1299  1/4/2021  267.054
1300  1/5/2021  269.195
1301  1/6/2021  271.696
1302  1/7/2021  273.003

[1303 rows x 2 columns]

In [42]:
import pandas as pd
df = pd.read_csv('US CPI.csv')
X = df.drop(columns = ['CPI'])
X
y = df['CPI']
y

0         9.800
1         9.800
2         9.800
3         9.800
4         9.700
         ...   
1298    264.877
1299    267.054
1300    269.195
1301    271.696
1302    273.003
Name: CPI, Length: 1303, dtype: float64

In [16]:
df_1 = pd.read_csv('data.csv')
df_1
MA_EndDate_graph = {2007:441854, 2008:603281, 2009:442131, 2017:506578, 2018:601167, 2019:597884}
Years = list(MA_EndDate_graph.keys())
Moving_Avg = list(MA_EndDate_graph.values())
plt.bar(Years,Moving_Avg)

<BarContainer object of 6 artists>

In [13]:
import pandas as pd
df_1 = pd.read_csv('data.csv')
df_1
X = df_1.drop(columns=[ 'MA'])
X
y = df_1['MA']
y

0      441854
1      441854
2      441854
3      441854
4      451583
        ...  
342    603281
343    601167
344    605637
345    599339
346    597884
Name: MA, Length: 347, dtype: int64

In [4]:
import pandas as pd
df_2 = pd.read_csv('usprice_cust.csv')
df_2

Median and Average Sales Prices of New Homes Sold in United States  \
0                                                 NaN                   
1                                         Annual Data                   
2                                                 NaN                   
3                                              Period                   
4                                                1963                   
..                                                ...                   
61                                               2020                   
62                                               2021                   
63                                                NaN                   
64                                                NaN                   
65           Note: The sales price includes the land.                   

   Unnamed: 1 Unnamed: 2  
0         NaN        NaN  
1         NaN        NaN  
2         NaN        NaN  
3     Median     Average  
4     $18,000    $19,300  
..        ...        ...  
61   $336,900   $391,900  
62   $392,900   $453,700  
63        NaN        NaN  
64        NaN        NaN  
65        NaN        NaN  

[66 rows x 3 columns]

In [5]:
df_2.describe()


Median and Average Sales Prices of New Homes Sold in United States  \
count                                                  62                   
unique                                                 62                   
top                                           Annual Data                   
freq                                                    1                   

       Unnamed: 1 Unnamed: 2  
count          60         60  
unique         59         59  
top      $120,000    $26,600  
freq            2          2

In [24]:
import pandas as pd
df_3 = pd.read_csv('housesSold.csv')
df_3
total_sold_2021 = 762
df_3.isna().any()

Houses Sold by Region    True
Unnamed: 1               True
Unnamed: 2               True
Unnamed: 3               True
Unnamed: 4               True
Unnamed: 5               True
dtype: bool

In [23]:
df_income = pd.read_csv('CPI_Report.csv')
df_income #UNnamed is first 12 months of the year, followed by Half1(Quarter1 and Quarter2) and Half2(Quarter3 and Quarter4)

Year      Jan      Feb      Mar      Apr      May      Jun      Jul  \
0   2000  168.800  169.800  171.200  171.300  171.500  172.400  172.800   
1   2001  175.100  175.800  176.200  176.900  177.700  178.000  177.500   
2   2002  177.100  177.800  178.800  179.800  179.800  179.900  180.100   
3   2003  181.700  183.100  184.200  183.800  183.500  183.700  183.900   
4   2004  185.200  186.200  187.400  188.000  189.100  189.700  189.400   
5   2005  190.700  191.800  193.300  194.600  194.400  194.500  195.400   
6   2006  198.300  198.700  199.800  201.500  202.500  202.900  203.500   
7   2007  202.416  203.499  205.352  206.686  207.949  208.352  208.299   
8   2008  211.080  211.693  213.528  214.823  216.632  218.815  219.964   
9   2009  211.143  212.193  212.709  213.240  213.856  215.693  215.351   
10  2010  216.687  216.741  217.631  218.009  218.178  217.965  218.011   
11  2011  220.223  221.309  223.467  224.906  225.964  225.722  225.922   
12  2012  226.665  227.663  229.392  230.085  229.815  229.478  229.104   
13  2013  230.280  232.166  232.773  232.531  232.945  233.504  233.596   
14  2014  233.916  234.781  236.293  237.072  237.900  238.343  238.250   
15  2015  233.707  234.722  236.119  236.599  237.805  238.638  238.654   
16  2016  236.916  237.111  238.132  239.261  240.229  241.018  240.628   
17  2017  242.839  243.603  243.801  244.524  244.733  244.955  244.786   
18  2018  247.867  248.991  249.554  250.546  251.588  251.989  252.006   
19  2019  251.712  252.776  254.202  255.548  256.092  256.143  256.571   
20  2020  257.971  258.678  258.115  256.389  256.394  257.797  259.101   
21  2021  261.582  263.014  264.877  267.054  269.195  271.696  273.003   
22  2022  281.148      NaN      NaN      NaN      NaN      NaN      NaN   

        Aug      Sep      Oct      Nov      Dec    HALF1    HALF2  
0   172.800  173.700  174.000  174.100  174.000  170.800  173.600  
1   177.500  178.300  177.700  177.400  176.700  176.600  177.500  
2   180.700  181.000  181.300  181.300  180.900  178.900  180.900  
3   184.600  185.200  185.000  184.500  184.300  183.300  184.600  
4   189.500  189.900  190.900  191.000  190.300  187.600  190.200  
5   196.400  198.800  199.200  197.600  196.800  193.200  197.400  
6   203.900  202.900  201.800  201.500  201.800  200.600  202.600  
7   207.917  208.490  208.936  210.177  210.036  205.709  208.976  
8   219.086  218.783  216.573  212.425  210.228  214.429  216.177  
9   215.834  215.969  216.177  216.330  215.949  213.139  215.935  
10  218.312  218.439  218.711  218.803  219.179  217.535  218.576  
11  226.545  226.889  226.421  226.230  225.672  223.598  226.280  
12  230.379  231.407  231.317  230.221  229.601  228.850  230.338  
13  233.877  234.149  233.546  233.069  233.049  232.366  233.548  
14  237.852  238.031  237.433  236.151  234.812  236.384  237.088  
15  238.316  237.945  237.838  237.336  236.525  236.265  237.769  
16  240.849  241.428  241.729  241.353  241.432  238.778  241.237  
17  245.519  246.819  246.663  246.669  246.524  244.076  246.163  
18  252.146  252.439  252.885  252.038  251.233  250.089  252.125  
19  256.558  256.759  257.346  257.208  256.974  254.412  256.903  
20  259.918  260.280  260.388  260.229  260.474  257.557  260.065  
21  273.567  274.310  276.589  277.948  278.802  266.236  275.703  
22      NaN      NaN      NaN      NaN      NaN      NaN      NaN

In [4]:
import pandas as pd

df_income = pd.read_csv('CPI_Report.csv')
df_income

Year      Jan      Feb      Mar      Apr      May      Jun      Jul  \
0   2000  168.800  169.800  171.200  171.300  171.500  172.400  172.800   
1   2001  175.100  175.800  176.200  176.900  177.700  178.000  177.500   
2   2002  177.100  177.800  178.800  179.800  179.800  179.900  180.100   
3   2003  181.700  183.100  184.200  183.800  183.500  183.700  183.900   
4   2004  185.200  186.200  187.400  188.000  189.100  189.700  189.400   
5   2005  190.700  191.800  193.300  194.600  194.400  194.500  195.400   
6   2006  198.300  198.700  199.800  201.500  202.500  202.900  203.500   
7   2007  202.416  203.499  205.352  206.686  207.949  208.352  208.299   
8   2008  211.080  211.693  213.528  214.823  216.632  218.815  219.964   
9   2009  211.143  212.193  212.709  213.240  213.856  215.693  215.351   
10  2010  216.687  216.741  217.631  218.009  218.178  217.965  218.011   
11  2011  220.223  221.309  223.467  224.906  225.964  225.722  225.922   
12  2012  226.665  227.663  229.392  230.085  229.815  229.478  229.104   
13  2013  230.280  232.166  232.773  232.531  232.945  233.504  233.596   
14  2014  233.916  234.781  236.293  237.072  237.900  238.343  238.250   
15  2015  233.707  234.722  236.119  236.599  237.805  238.638  238.654   
16  2016  236.916  237.111  238.132  239.261  240.229  241.018  240.628   
17  2017  242.839  243.603  243.801  244.524  244.733  244.955  244.786   
18  2018  247.867  248.991  249.554  250.546  251.588  251.989  252.006   
19  2019  251.712  252.776  254.202  255.548  256.092  256.143  256.571   
20  2020  257.971  258.678  258.115  256.389  256.394  257.797  259.101   
21  2021  261.582  263.014  264.877  267.054  269.195  271.696  273.003   
22  2022  281.148      NaN      NaN      NaN      NaN      NaN      NaN   

        Aug      Sep      Oct      Nov      Dec    HALF1    HALF2  
0   172.800  173.700  174.000  174.100  174.000  170.800  173.600  
1   177.500  178.300  177.700  177.400  176.700  176.600  177.500  
2   180.700  181.000  181.300  181.300  180.900  178.900  180.900  
3   184.600  185.200  185.000  184.500  184.300  183.300  184.600  
4   189.500  189.900  190.900  191.000  190.300  187.600  190.200  
5   196.400  198.800  199.200  197.600  196.800  193.200  197.400  
6   203.900  202.900  201.800  201.500  201.800  200.600  202.600  
7   207.917  208.490  208.936  210.177  210.036  205.709  208.976  
8   219.086  218.783  216.573  212.425  210.228  214.429  216.177  
9   215.834  215.969  216.177  216.330  215.949  213.139  215.935  
10  218.312  218.439  218.711  218.803  219.179  217.535  218.576  
11  226.545  226.889  226.421  226.230  225.672  223.598  226.280  
12  230.379  231.407  231.317  230.221  229.601  228.850  230.338  
13  233.877  234.149  233.546  233.069  233.049  232.366  233.548  
14  237.852  238.031  237.433  236.151  234.812  236.384  237.088  
15  238.316  237.945  237.838  237.336  236.525  236.265  237.769  
16  240.849  241.428  241.729  241.353  241.432  238.778  241.237  
17  245.519  246.819  246.663  246.669  246.524  244.076  246.163  
18  252.146  252.439  252.885  252.038  251.233  250.089  252.125  
19  256.558  256.759  257.346  257.208  256.974  254.412  256.903  
20  259.918  260.280  260.388  260.229  260.474  257.557  260.065  
21  273.567  274.310  276.589  277.948  278.802  266.236  275.703  
22      NaN      NaN      NaN      NaN      NaN      NaN      NaN

In [6]:
from sklearn.tree import DecisionTreeClassifier
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
df_income = pd.read_csv('CPI_Report.csv')
X = df_income.drop(columns=[ 'HALF2'])
X
yearly_CPI_2000_2021 = df_income['HALF2']
yearly_CPI_2000_2021


0     173.600
1     177.500
2     180.900
3     184.600
4     190.200
5     197.400
6     202.600
7     208.976
8     216.177
9     215.935
10    218.576
11    226.280
12    230.338
13    233.548
14    237.088
15    237.769
16    241.237
17    246.163
18    252.125
19    256.903
20    260.065
21    275.703
22        NaN
Name: HALF2, dtype: float64

In [109]:
df_4 = pd.read_csv('US CPI.csv')
df_4
df_4.tail()

Date      CPI
1298  1/3/2021  264.877
1299  1/4/2021  267.054
1300  1/5/2021  269.195
1301  1/6/2021  271.696
1302  1/7/2021  273.003

In [11]:
import pandas as pd
df = pd.read_csv('US CPI.csv')
df
X = df.drop(columns = ['CPI'])
X
y = df['CPI']
y

0         9.800
1         9.800
2         9.800
3         9.800
4         9.700
         ...   
1298    264.877
1299    267.054
1300    269.195
1301    271.696
1302    273.003
Name: CPI, Length: 1303, dtype: float64

In [17]:
df_book = pd.read_csv('Book1.csv')
df_book

Nation  High CPI  Inflation Rate                CPI  Year  \
0         Turkey         1           48.69              760.5  2022   
1            USA         1            7.50            280.148  2022   
2      Venezuela         1         1198.00  2,750,974,554,036  2022   
3          Sudan         1          340.00               9138  2021   
4       Zimbabwe         1           60.70               6490  2021   
5          Syria         1          139.00               2668  2022   
6       Suriname         1           63.30             775.92  2021   
7      Argentina         1           51.20             605.03  2022   
8           Iran         1           35.20              388.2  2022   
9       Ethiopia         1           33.00             398.65  2022   
10        Rwanda         0           -2.00              163.7  2022   
11          Chad         0           -0.05             123.42  2021   
12         Gabon         0            0.60              87.53  2022   
13         Japan         0            0.60              100.3  2022   
14       Bahrain         0            0.70             116.04  2022   
15          Fiji         0            1.80             128.07  2021   
16       Bolivia         0            0.90             151.29  2022   
17  Saudi Arabia         0            1.10              85.64  2022   

   Financial Status  
0          Negative  
1          Negative  
2          Negative  
3          Negative  
4          Negative  
5          Negative  
6          Negative  
7          Negative  
8          Negative  
9          Negative  
10         Positive  
11         Positive  
12         Positive  
13         Positive  
14         Positive  
15         Positive  
16         Positive  
17         Positive

In [22]:
X = df_book.drop(columns=['Inflation Rate'])
X
y = df_book['Inflation Rate']
y

0       48.69
1        7.50
2     1198.00
3      340.00
4       60.70
5      139.00
6       63.30
7       51.20
8       35.20
9       33.00
10      -2.00
11      -0.05
12       0.60
13       0.60
14       0.70
15       1.80
16       0.90
17       1.10
Name: Inflation Rate, dtype: float64

In [20]:
X = df_book.drop(columns=['CPI'])
X
y = df_book['CPI']
y

0                 760.5
1               280.148
2     2,750,974,554,036
3                  9138
4                  6490
5                  2668
6                775.92
7                605.03
8                 388.2
9                398.65
10                163.7
11               123.42
12                87.53
13                100.3
14               116.04
15               128.07
16               151.29
17                85.64
Name: CPI, dtype: object

from sklearn.tree import DecisionTreeClassifier
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
df_book_inflation_rate_by_country = pd.read_csv('Book1.csv')
df_book_inflation_rates_by_country


Infl_rate_CPI = {48.69:760.5, 7.50:280.148, 1198:2750974554036,51.20:605.03,1.10:85.64, 33.00:398.65, 0.60:87.53, 0.60:100.3}
Rate = list(Infl_rate_CPI.keys())
CPI = list(Infl_rate_CPI.values())
plt.bar(Rate,CPI)
plt.show()


In [9]:
 %pylab
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import datetime
df_cpi_report = pd.read_csv('CPI_Report.csv')
df_cpi_report
df_cpi_report.head()

Using matplotlib backend: <object object at 0x0000024AD29E9450>
%pylab is deprecated, use %matplotlib inline and import the required libraries.
Populating the interactive namespace from numpy and matplotlib


Year    Jan    Feb    Mar    Apr    May    Jun    Jul    Aug    Sep    Oct  \
0  2000  168.8  169.8  171.2  171.3  171.5  172.4  172.8  172.8  173.7  174.0   
1  2001  175.1  175.8  176.2  176.9  177.7  178.0  177.5  177.5  178.3  177.7   
2  2002  177.1  177.8  178.8  179.8  179.8  179.9  180.1  180.7  181.0  181.3   
3  2003  181.7  183.1  184.2  183.8  183.5  183.7  183.9  184.6  185.2  185.0   
4  2004  185.2  186.2  187.4  188.0  189.1  189.7  189.4  189.5  189.9  190.9   

     Nov    Dec  HALF1  HALF2  
0  174.1  174.0  170.8  173.6  
1  177.4  176.7  176.6  177.5  
2  181.3  180.9  178.9  180.9  
3  184.5  184.3  183.3  184.6  
4  191.0  190.3  187.6  190.2

In [11]:
df_cpi_report.tail()

Year      Jan      Feb      Mar      Apr      May      Jun      Jul  \
18  2018  247.867  248.991  249.554  250.546  251.588  251.989  252.006   
19  2019  251.712  252.776  254.202  255.548  256.092  256.143  256.571   
20  2020  257.971  258.678  258.115  256.389  256.394  257.797  259.101   
21  2021  261.582  263.014  264.877  267.054  269.195  271.696  273.003   
22  2022  281.148      NaN      NaN      NaN      NaN      NaN      NaN   

        Aug      Sep      Oct      Nov      Dec    HALF1    HALF2  
18  252.146  252.439  252.885  252.038  251.233  250.089  252.125  
19  256.558  256.759  257.346  257.208  256.974  254.412  256.903  
20  259.918  260.280  260.388  260.229  260.474  257.557  260.065  
21  273.567  274.310  276.589  277.948  278.802  266.236  275.703  
22      NaN      NaN      NaN      NaN      NaN      NaN      NaN

In [6]:
df_cpi_report['Jan'].plot(figsize=(22,2))

<AxesSubplot:>

In [7]:
df_cpi_report.rolling(window=21).mean()['HALF2'].plot()

<AxesSubplot:>

In [11]:
import pandas as pd
dataset_test = pd.read_csv('CPI_Report.csv')
dataset_test

Year      Jan      Feb      Mar      Apr      May      Jun      Jul  \
0   2000  168.800  169.800  171.200  171.300  171.500  172.400  172.800   
1   2001  175.100  175.800  176.200  176.900  177.700  178.000  177.500   
2   2002  177.100  177.800  178.800  179.800  179.800  179.900  180.100   
3   2003  181.700  183.100  184.200  183.800  183.500  183.700  183.900   
4   2004  185.200  186.200  187.400  188.000  189.100  189.700  189.400   
5   2005  190.700  191.800  193.300  194.600  194.400  194.500  195.400   
6   2006  198.300  198.700  199.800  201.500  202.500  202.900  203.500   
7   2007  202.416  203.499  205.352  206.686  207.949  208.352  208.299   
8   2008  211.080  211.693  213.528  214.823  216.632  218.815  219.964   
9   2009  211.143  212.193  212.709  213.240  213.856  215.693  215.351   
10  2010  216.687  216.741  217.631  218.009  218.178  217.965  218.011   
11  2011  220.223  221.309  223.467  224.906  225.964  225.722  225.922   
12  2012  226.665  227.663  229.392  230.085  229.815  229.478  229.104   
13  2013  230.280  232.166  232.773  232.531  232.945  233.504  233.596   
14  2014  233.916  234.781  236.293  237.072  237.900  238.343  238.250   
15  2015  233.707  234.722  236.119  236.599  237.805  238.638  238.654   
16  2016  236.916  237.111  238.132  239.261  240.229  241.018  240.628   
17  2017  242.839  243.603  243.801  244.524  244.733  244.955  244.786   
18  2018  247.867  248.991  249.554  250.546  251.588  251.989  252.006   
19  2019  251.712  252.776  254.202  255.548  256.092  256.143  256.571   
20  2020  257.971  258.678  258.115  256.389  256.394  257.797  259.101   
21  2021  261.582  263.014  264.877  267.054  269.195  271.696  273.003   
22  2022  281.148      NaN      NaN      NaN      NaN      NaN      NaN   

        Aug      Sep      Oct      Nov      Dec    HALF1    HALF2  
0   172.800  173.700  174.000  174.100  174.000  170.800  173.600  
1   177.500  178.300  177.700  177.400  176.700  176.600  177.500  
2   180.700  181.000  181.300  181.300  180.900  178.900  180.900  
3   184.600  185.200  185.000  184.500  184.300  183.300  184.600  
4   189.500  189.900  190.900  191.000  190.300  187.600  190.200  
5   196.400  198.800  199.200  197.600  196.800  193.200  197.400  
6   203.900  202.900  201.800  201.500  201.800  200.600  202.600  
7   207.917  208.490  208.936  210.177  210.036  205.709  208.976  
8   219.086  218.783  216.573  212.425  210.228  214.429  216.177  
9   215.834  215.969  216.177  216.330  215.949  213.139  215.935  
10  218.312  218.439  218.711  218.803  219.179  217.535  218.576  
11  226.545  226.889  226.421  226.230  225.672  223.598  226.280  
12  230.379  231.407  231.317  230.221  229.601  228.850  230.338  
13  233.877  234.149  233.546  233.069  233.049  232.366  233.548  
14  237.852  238.031  237.433  236.151  234.812  236.384  237.088  
15  238.316  237.945  237.838  237.336  236.525  236.265  237.769  
16  240.849  241.428  241.729  241.353  241.432  238.778  241.237  
17  245.519  246.819  246.663  246.669  246.524  244.076  246.163  
18  252.146  252.439  252.885  252.038  251.233  250.089  252.125  
19  256.558  256.759  257.346  257.208  256.974  254.412  256.903  
20  259.918  260.280  260.388  260.229  260.474  257.557  260.065  
21  273.567  274.310  276.589  277.948  278.802  266.236  275.703  
22      NaN      NaN      NaN      NaN      NaN      NaN      NaN

In [83]:
dataset_test.head()

Year    Jan    Feb    Mar    Apr    May    Jun    Jul    Aug    Sep    Oct  \
0  2000  168.8  169.8  171.2  171.3  171.5  172.4  172.8  172.8  173.7  174.0   
1  2001  175.1  175.8  176.2  176.9  177.7  178.0  177.5  177.5  178.3  177.7   
2  2002  177.1  177.8  178.8  179.8  179.8  179.9  180.1  180.7  181.0  181.3   
3  2003  181.7  183.1  184.2  183.8  183.5  183.7  183.9  184.6  185.2  185.0   
4  2004  185.2  186.2  187.4  188.0  189.1  189.7  189.4  189.5  189.9  190.9   

     Nov    Dec  HALF1  HALF2  
0  174.1  174.0  170.8  173.6  
1  177.4  176.7  176.6  177.5  
2  181.3  180.9  178.9  180.9  
3  184.5  184.3  183.3  184.6  
4  191.0  190.3  187.6  190.2

In [84]:
dataset_test.tail()

Year      Jan      Feb      Mar      Apr      May      Jun      Jul  \
18  2018  247.867  248.991  249.554  250.546  251.588  251.989  252.006   
19  2019  251.712  252.776  254.202  255.548  256.092  256.143  256.571   
20  2020  257.971  258.678  258.115  256.389  256.394  257.797  259.101   
21  2021  261.582  263.014  264.877  267.054  269.195  271.696  273.003   
22  2022  281.148      NaN      NaN      NaN      NaN      NaN      NaN   

        Aug      Sep      Oct      Nov      Dec    HALF1    HALF2  
18  252.146  252.439  252.885  252.038  251.233  250.089  252.125  
19  256.558  256.759  257.346  257.208  256.974  254.412  256.903  
20  259.918  260.280  260.388  260.229  260.474  257.557  260.065  
21  273.567  274.310  276.589  277.948  278.802  266.236  275.703  
22      NaN      NaN      NaN      NaN      NaN      NaN      NaN

In [86]:
dataset_test["Year"]=dataset_test["Year"].replace(',', '').astype(float)

In [87]:
dataset_test["HALF2"]=dataset_test["HALF2"].replace(',', '').astype(float)

In [88]:
test_set = dataset_test["Jan"]
test_set = pd.DataFrame(test_set)

In [89]:
test_set.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23 entries, 0 to 22
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Jan     23 non-null     float64
dtypes: float64(1)
memory usage: 312.0 bytes


In [90]:
dataset_total = pd.concat((df_cpi_report['Jan'], dataset_test['Jan']), axis=0)
inputs